src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -99,30 +53,41 @@ internal TagHelperAttributeDescriptor([NotNull] string name, [NotNull] PropertyI
         /// to allow empty values for HTML attributes matching this <see cref="TagHelperAttributeDescriptor"/>. If
         /// <c>false</c> empty values for such matching attributes lead to errors.
         /// </value>
-        public bool IsStringProperty { get; }
+        public bool IsStringProperty { get; private set; }
 
         /// <summary>
         /// The HTML attribute name or, if <see cref="IsIndexer"/> is <c>true</c>, the prefix for matching attribute
         /// names.
         /// </summary>
-        public string Name { get; }
+        public string Name { get; [param: NotNull] set; }
 
         /// <summary>
         /// The name of the CLR property that corresponds to the HTML attribute.
         /// </summary>
-        public string PropertyName { get; }
+        public string PropertyName { get; [param: NotNull] set; }
```

Comment:
Note that nothing guarantees `Name`, `PropertyName`, or `TypeName` are ever set. Fortunately these descriptors are originally created in `TagHelperDescriptorFactory` -- after checking source attribute values. But may be worth adding `Debug.Assert()` expressions checking these values in `TagHelperBlockRewriter` or some place similar.